py3status/modules/volume_status.py

```diff
@@ -387,14 +387,26 @@ def volume_status(self):
         perc, muted = self.backend.get_volume()
 
         color = None
-        if muted:
-            color = self.py3.COLOR_MUTED or self.py3.COLOR_BAD
-        if not self.py3.is_color(color):
-            # determine the color based on the current volume level
-            color = self._perc_to_color(perc)
-
-        # format the output
-        text = self._format_output(self.format_muted if muted else self.format, perc)
+        if perc is None or muted is None:
+            color = self.py3.COLOR_BAD
+            perc = "?"
+            if hasattr(self, "format_missing"):
+                text = self._format_output(self.format_missing, perc)
```

Comment:
Can we not introduce a hidden `format_missing` config? Also, because of this one thing, we add bunch of things. Idk what to do here as I'd want to simplify things and put them in the `format` and use the formatter instead. The future is uncertainty.

Right now in any `format` config, the `perc` could stay `None` to display `None` or if users want, `[{percent}|\?show ?]` to display `?`... which could be same as `format_missing`. Also, I think nobody would want to customize `format_missing` because they might prefer to have a working `volume_status`.

It's same for `muted = None` too.